factory/peerSignatureHandler/peerSignatureHandler.go

```diff
@@ -5,7 +5,7 @@ import (
 
 	"github.com/ElrondNetwork/elrond-go-core/core"
 	"github.com/ElrondNetwork/elrond-go-core/core/check"
-	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go-crypto"
```

Comment:
this file looks more like a processor with cacher. Could have been moved to process. 
Can remain where it is for now.